src/components/ErrorAlert/index.js

```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+function Error() {
+    return <div>Error retrieving top market cap coins!</div>;
```

Comment:
This is ok..but you're going to be re-using this `error` component for all of your api calls error handling. So you should be passing in the message, like i originally walked through with you. 

`message` should be a prop passed in from the parent components. 

`message` should also be used to conditionally render the error component from within. Instead of checking the state at the parent like you have 
```
{error ? <Error /> : null }
```

The check should look something like this, inside of this component:

```
function Error({message}) {
if (!message) return null

    return <div>{message}</div>;
```

This will also require that you dynamically pass the message prop down from the parent. So you'd do something like this
```
<Error message={error ? 'ERROR MESSAGE' : null} />
```